redfish-core/lib/assembly.hpp

```diff
@@ -711,6 +712,224 @@ inline void getChassis(const std::shared_ptr<bmcweb::AsyncResp>& aResp,
+                return;
+            }
+
+            std::vector<std::string> assemblyAssoc;
+            for (const auto& association : *value)
+            {
+                if (std::get<0>(association) != "assembly")
+                {
+                    continue;
+                }
+                assemblyAssoc.emplace_back(std::get<2>(association));
+            }
+
+            if (assemblyAssoc.empty())
+            {
+                BMCWEB_LOG_ERROR
+                    << "No assembly associations in the ["
+                    << assemblyParentObjPath.str
+                    << "] to fill Assembly id of the assembled object ["
+                    << assembledObjPath.str << "]";
+                messages::internalError(aResp->res);
+                return;
+            }
+
+            // Mak sure whether the retrieved assembly associations are
+            // implemented before finding the assembly id as per bmcweb
+            // Assembly design.
+            crow::connections::systemBus->async_method_call(
+                [aResp, assemblyUriPropPath, assemblyParentObjPath,
+                 assembledObjPath, assemblyAssoc, assembledUriVal](
+                    const boost::system::error_code ec,
+                    const std::vector<
+                        std::pair<std::string,
+                                  std::vector<std::pair<
+                                      std::string, std::vector<std::string>>>>>&
+                        subtrees) {
+                    if (ec)
+                    {
+                        BMCWEB_LOG_ERROR
+                            << "DBUS response error [" << ec.value() << " : "
+                            << ec.message()
+                            << "] when tried to get the subtree to check "
+                            << "assembled objects implementation of the ["
+                            << assemblyParentObjPath.str
+                            << "] to find assembled object id of the ["
+                            << assembledObjPath.str
+                            << "] to fill in the URI property";
+                        messages::internalError(aResp->res);
+                        return;
+                    }
+
+                    if (subtrees.empty())
+                    {
+                        BMCWEB_LOG_ERROR
+                            << "No subtree in the ["
+                            << assemblyParentObjPath.str
+                            << "] to check assembled objects implementation "
+                            << "to fill the assembled object [ "
+                            << assembledObjPath.str
+                            << "] id in the URI property";
+                        messages::internalError(aResp->res);
+                        return;
+                    }
+
+                    std::vector<std::string> implAssemblyAssocs;
+                    for (const auto& subtree : subtrees)
```

Comment:
What if you found more than 1 object?

Reply:
Should you do something more like 
https://github.com/openbmc/bmcweb/blob/55f79e6fe156505cdaddd521212a29b2a977193a/redfish-core/lib/systems.hpp#L2560

Reply:
Here we will get more (say for example vrm) and each vrm object is treated as a separate assembly id, right?

```
busctl -j call xyz.openbmc_project.ObjectMapper /xyz/openbmc_project/object_mapper xyz.openbmc_project.ObjectMapper GetSubTree sias /xyz/openbmc_project/inventory 0 9 xyz.openbmc_project.Inventory.Item.Vrm xyz.openbmc_project.Inventory.Item.Tpm xyz.openbmc_project.Inventory.Item.Panel xyz.openbmc_project.Inventory.Item.Battery xyz.openbmc_project.Inventory.Item.DiskBackplane xyz.openbmc_project.Inventory.Item.Board xyz.openbmc_project.Inventory.Item.Connector xyz.openbmc_project.Inventory.Item.Drive xyz.openbmc_project.Inventory.Item.Board.Motherboard
```